yjit/src/asm/mod.rs

```diff
@@ -36,14 +40,20 @@ struct LabelRef {
 /// Block of memory into which instructions can be assembled
 pub struct CodeBlock {
     // Memory for storing the encoded instructions
-    mem_block: VirtualMem,
+    mem_block: Rc<RefCell<VirtualMem>>,
 
     // Memory block size
     mem_size: usize,
 
     // Current writing position
     write_pos: usize,
 
+    // TODO: comment
+    page_size: usize,
+
+    // TODO: comment
```

Comment:
I know we all have different ways of working but I find it helpful to write the comment first, sometimes before even writing any code. It's kind of like [rubber duck debugging](https://en.wikipedia.org/wiki/Rubber_duck_debugging).

Reply:
yeah right, I was actually doing that at first, but because this PR required many times of design changes behind the scenes, I stopped doing that to save some time. I ran out of time today for debugging, but I'll try to write them soon.